src/store/api.js

```diff
@@ -2,9 +2,64 @@ export const XDAI_SUBGRAPH_URL = process.env.REACT_APP_XDAI_SUBGRAPH_URL;
 export const MAINNET_SUBGRAPH_URL = process.env.REACT_APP_MAINNET_SUBGRAPH_URL;
 export const POAP_API_URL = process.env.REACT_APP_POAP_API_URL;
 export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+export const OrderType = {
+  id: {
+    name: 'Id',
+    val: 'id'
+  },
+  tokenCount: {
+    name: 'Supply',
+    val: 'tokenCount'
+  },
+  transferCount: {
+    name: 'Transfers',
+    val: 'transferCount'
+  },
+  date: {
+    name: 'Date',
+    val: 'start_date'
+  },
+  city: {
+    name: 'City',
+    val: 'city'
+  },
+}
+export const OrderDirection = {
+  ascending: {
+    name: 'Ascending',
+    val: 'asc'
+  },
+  descending: {
+    name: 'Descending',
+    val: 'desc'
+  },
+}
 
-export async function getEvents() {
-  const res = await fetch(`${POAP_API_URL}/events`)
+export const PAGE_LIMIT = 20
+
+export async function getPaginatedEvents({name = undefined, event_ids = undefined,
+                                           offset = undefined, limit = undefined,
+                                           orderBy = undefined, privateEvents = undefined}) {
+  let url = `${POAP_API_URL}/paginated-events?`
+  if (name) {
+    url += `&name=${name}`
+  }
+  if (event_ids && event_ids.length) {
+    url += `&event_ids=${event_ids}`
+  }
+  if (limit && limit > 0) {
+    url += `&limit=${limit}`
+  }
+  if (offset !== undefined && offset >= 0) {
+    url += `&offset=${offset}`
+  }
+  if (orderBy?.type && orderBy?.order) {
+    url += `&sort_field=${orderBy.type}&sort_dir=${orderBy.order}`
+  }
+  if (privateEvents !== undefined) {
+    url += `&private_event=${privateEvents}`
+  }
```

Comment:
For security reasons we should use the URL API (https://developer.mozilla.org/en-US/docs/Web/API/URL) and build the whole URL with that including the query params